### PR TITLE
api: proposal proxy ports

### PR DESCRIPTION
**Pull request by zirain**
this's an alternative for https://github.com/envoyproxy/gateway/pull/1787
xref: https://github.com/envoyproxy/gateway/issues/1703
follow up of: https://github.com/envoyproxy/gateway/pull/2015#discussion_r1366079885

### CODE REVIEW

**Review by qicz — APPROVED** [review]

**Comment by arkodg** [comment]
@zirain can we add this post v0.6 ?

**Comment by zirain** [comment]
> @zirain can we add this post v0.6 ?

sure, it's not urgent.

File: api/v1alpha1/envoyproxy_types.go [hunk 1]
```diff
@@ -73,6 +79,20 @@ type EnvoyProxySpec struct {
...
+	// AdminPort defines the port for the Proxy admin interface.
+	// If unspecified, defaults to 19000.
+	// +optional
+	AdminPort *int32 `json:"adminPort,omitempty"`
+
+	// ExtraPort defines the extra port for the Proxy extra interface like
+	// HealthCheck, Prometheus metrics.
+	// If unspecified, defaults to 19001.
+	// +optional
+	ExtraPort *int32 `json:"extraPort,omitempty"`
```
Comment (arkodg) [inline api/v1alpha1/envoyproxy_types.go:93]: `extraPort` is not a great name imo
should we split this up into `prometheusPort` & `healthCheckPort` ?

Comment (zirain) [reply api/v1alpha1/envoyproxy_types.go:93]: nop, it will/should shared one port in most cases.

Comment (arkodg) [reply api/v1alpha1/envoyproxy_types.go:93]: can we then come up with a better name

Comment (zirain) [reply api/v1alpha1/envoyproxy_types.go:93]: do you have any good idea?

**Pull request closed without merging**